src/irr/asset/CGLILoader.cpp

```diff
@@ -150,6 +150,8 @@ namespace irr
 					region->imageExtent.width = texture.extent(regionIndex).x;
 					region->imageExtent.height = texture.extent(regionIndex).y;
 					region->imageExtent.depth = texture.extent(regionIndex).z;
+					region->bufferRowLength = IImageAssetHandlerBase::calcPitchInBlocks(region->imageExtent.width, texelBlockByteSize / asset::getBlockDimensions(imageInfo.format).X);
```

Comment:
this needs logical fixing

Reply:
width is in texels

Reply:
since texelBlockByteSize may be in texels or blocks, we need to make sure byte size will be in texels , input may be various

Reply:
to final calculate row pitch in texels

Reply:
this is plain wrong, `bufferRowLength` is supposed to equal `region->imageExtent.width` rounded up to the multiple of block size (not bytesize, physical dimensions)

Reply:
the logic is still completely and utterly fucked

Reply:
"done" XD